xrtpy/util/xrt_remove_lightleak.py

```diff
@@ -0,0 +1,275 @@
+        will be subtracted from
+
+    kfact : float, default: 1.0
+        k-factor to apply when subtracting the light leak image:
+        ``out_data = in_data - k * [leak_img]``
+
+
+    leak_image : float array, dimensions 1024x1024,  optional
+        A leak image to subtract, if a non-default image is desired.
+        It's assumed that the image is 1024x1024, prepped and exposure
+        normalized.
+
+    verbose : `bool`, default: `False`
+        If `True`, print out extra messages.
+
+    Returns:
+    --------
+    out_map : ~sunpy.map.sources.hinode.XRTMap
+        |Map| of input image with the light leak removed. The metadata HISTORY
+        is also updated to reflect the fact that the light leak was removed.
+
+    Example:
+    --------
+    >>> file = ``"comp_XRT20200220_061539.6.fits"`` # doctest: +SKIP
+    >>> in_map = Map(file) # doctest: +SKIP
+    >>> out_map =xrt_remove_lightleak(in_map) # doctest: +SKIP
+
+    Notes:
+    ------
+    (Taken from the IDL routine :file:`xrt_synleaksub.pro`)
+    1. XRT images obtained after 9-May-2021 suffer visible stray light
+    contamination (light leak) due to the pre-filter failure, i.e.  tiny
+    rupture development occurred multiple times as follows:
+
+    phase 1 :  9-May-2012 12:00
+    phase 2 : 14-Jun-2015 12:30
+    phase 3 : 27-May-2017 11:00
+    phase 4 : 29-May-2018 00:00
+    phase 5 :  8-Jun-2022 12:40
+
+    2. The light leak correction is done by simply subtracting the light leak
+    image (visible stray light component included in each X-ray filter pair)
+    obtained during the Hinode satellite's eclipse season, which occurs
+    roughly May to August each year.  The light leak image has the following
+    characteristics:
+
+    * Leak pattern and intensity differ with each filter and also with the
+      satellite pointing.
+    * Leak pattern and intensity of the same filter and pointing are roughly
+      constant during each stray light phase (above), but vary by ~10%
+      depending on the growth of the contamination layer on the CCD (that
+      repeats gradual increase and jump down after CCD bakeouts).
+
+    3. The image for the light leak correction should therefore be selected
+    for the same filter, same satellite pointing, same stray light phase then
+    ideally be adjusted for the level of CCD contamination at the time of
+    observation.  However, as a practical matter, preparing the leak image for
+    every possible pointing is hard to achieve, while we have good amount of
+    light leak measurements at the disk center pointing. The full-disk
+    composite images are therefore corrected most reliably for the light leak.
+
+    4. Intensity variation of the leak image due to the growth of CCD
+    contamination layer is well determined for Ti_poly at the stray light
+    phase 1 by using the intensity correlation between Ti_poly and Al_mesh
+    images (cg. Takeda et al. 2016, SolPhys.  291, p.317). The resulting
```

Comment:
```suggestion
    images (cg. :cite:p:`Takeda:2016`, p.317). The resulting
```
...would also need to add this to `docs/bibliography.bib`.

Reply:
Yes, I will take care of this now. Thanks for pointing it out.